diff --git a/src/plugins/imageformats/tga/qtgafile.cpp b/src/plugins/imageformats/tga/qtgafile.cpp
index f1b9af3..13eb153 100644
--- a/src/plugins/imageformats/tga/qtgafile.cpp
+++ b/src/plugins/imageformats/tga/qtgafile.cpp
@@ -135,25 +135,6 @@ QTgaFile::QTgaFile(QIODevice *device)
         mErrorMessage = tr("Image size exceeds limit");
         return;
     }
-    int curPos = mDevice->pos();
-    int fileBytes = mDevice->size();
-    if (!mDevice->seek(fileBytes - FooterSize))
-    {
-        mErrorMessage = tr("Could not seek to image read footer");
-        return;
-    }
-    char footer[FooterSize];
-    if (mDevice->read(reinterpret_cast<char*>(footer), FooterSize) != FooterSize) {
-        mErrorMessage = tr("Could not read footer");
-    }
-    if (qstrncmp(&footer[SignatureOffset], "TRUEVISION-XFILE", 16) != 0)
-    {
-        mErrorMessage = tr("Image type (non-TrueVision 2.0) not supported");
-    }
-    if (!mDevice->seek(curPos))
-    {
-        mErrorMessage = tr("Could not reset to read data");
-    }
 }
 
 /*!
